packages/client/src/Stream.ts

```diff
@@ -203,25 +203,27 @@ class StreamrStream implements StreamMetadata {
         return this._streamRegistry.hasPublicPermission(this.id, permission)
     }
 
-    // async hasUserPermissions(operations: StreamPermission[], userId: string) {
-    //     this.assertUserId(userId)
-    //     const permissions = this._streamRegistry.getPermissionsForUser(this.id, userId, )
-    //     return this.hasPermissions(operations, userId)
-    // }
-    // async hasPermissions(operations: StreamPermission[], userId: string) {
-    //     const permissions = await this._streamRegistry.getPermissionsForUser(this.id, userId)
-
-    //     if (operation === StreamPermission.PUBLISH || operation === StreamPermission.SUBSCRIBE) {
-    //         return permissions[operation].gt(Date.now())
-    //     }
-    //     return permissions[operation]
-    // }
-
-    // async hasPermission(operation: StreamPermission, userId: string|undefined) {
-    //     const permissions = await this.hasPermissions([operation], userId)
-    //     if (!Array.isArray(permissions) || !permissions.length) { return undefined }
-    //     return permissions[0]
-    // }
+    async hasUserPermissions(requestedPermissions: StreamPermission[], userId: string): Promise<boolean> {
+        this.assertUserId(userId)
+        const permissions = await this._streamRegistry.getDirectPermissionsForUser(this.id, userId)
+        for (const requestedPermission of requestedPermissions) {
+            if (!permissions[requestedPermission]) {
+                return false
+            }
+        }
+        return true
+    }
+
+    async hasPermissions(requestedPermissions: StreamPermission[], userId: string): Promise<boolean> {
+        this.assertUserId(userId)
+        const permissions = await this._streamRegistry.getPermissionsForUser(this.id, userId)
+        for (const requestedPermission of requestedPermissions) {
```

Comment:
same here

Reply:
For code re-use, you could extract the common functionality of methods  `hasUserPermissions` and `hasPermissions` as a private helper method, something like the below:

```ts
private containsEveryRequestedPermission(requestedPermissions: StreamPermission[], fetchedPermissions: StreamPermission[]): boolean {
    // or alternatively what Tim suggested
    for (const requestedPermission of requestedPermissions) {
        if (!fetchedPermissions[requestedPermission]) {
            return false
        }
    }
    return true
}
```